src/main/java/com/zombiecastlerush/role/Role.java

```diff
@@ -96,36 +101,44 @@ public void setName(String name) {
         this.name = name;
     }
 
-    // Inventory methods
-    public Item pickUp(Item item) {
-        for (Item existingItem: this.getCurrentPosition().inventory.getItems()) {
-            System.out.println(existingItem.getName());
-            if (item.equals(existingItem)) {
-                this.inventory.addItems(item);
-                this.getCurrentPosition().inventory.deleteItems(item);
-                System.out.println(item.getName() + " picked up by " + this.getName());
-                return item;
+    public Inventory getInventory() {
+        return inventory;
+    }
+
+    public void setInventory(Inventory inventory) {
```

Comment:
The `setInventory` feels unnecessary if you've initialized it in the instance variables. You'll never need to set a reference to another inventory, each Room and Role will always start with a new Inventory(), no?

Reply:
removed